pkgs/applications/misc/cpp-ethereum/default.nix

```diff
@@ -27,7 +27,7 @@ stdenv.mkDerivation rec {
     sha256 = "1fxgpqhmjhpv0zzs1m3yf9h8mh25dqpa7pmcfy7f9qiqpfdr4zq9";
   };
 
-  cmakeFlags = with stdenv.lib; concatStringsSep " " ([ "-DCMAKE_BUILD_TYPE=Release" ] ++ extraCmakeFlags);
+  cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release" extraCmakeFlags ];
```

Comment:
The default option provided by Nix gets overridden in this case assuming the `nix-env` version is `1.11.4`. The source file `cmake/EthOptions.cmake` contains the following code in the definition of `macro(configure_project)`:
```cmake
if (NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING
    "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
endif()
```

If the string `"-DCMAKE_BUILD_TYPE=Release"` in the `cmakeFlags` list in `default.nix` is dropped, then the build process selects `RelWithDebInfo` and the log contains the lines:
```
------------------------------------------------------------------------
-- Configuring cpp-ethereum
------------------------------------------------------------------------
--                  CMake Version                            3.6.0
-- CMAKE_BUILD_TYPE Build type                               RelWithDebInfo
-- TARGET_PLATFORM  Target platform                          Linux
```
If the string `"-DCMAKE_BUILD_TYPE=Release"` is present, then the build log yields:
```
------------------------------------------------------------------------
-- Configuring cpp-ethereum
------------------------------------------------------------------------
--                  CMake Version                            3.6.0
-- CMAKE_BUILD_TYPE Build type                               Release
-- TARGET_PLATFORM  Target platform                          Linux
```


To verify that the resulting binaries are different, one can run:
```bash
readelf --debug-dump=line $(readlink -f $(which eth))
```
where `eth` is one of the executables created in the `bin` directory of the `cpp_ethereum` package. For the `Release` build the output is relatively short, and in case of `RelWithDebInfo` the output is very long.